### PR TITLE
Add documentation for the network interface used by the clients

### PR DESCRIPTION
**Pull request by skaupper**
The documentation will be embedded in header of `NetworkListener.h`

### CODE REVIEW

**Comment by skaupper** [comment]
Related trello card [here](https://trello.com/c/LeyHLyKN/7-2-dokumentation-der-netzwerkschnittstelle)

**Comment by skaupper** [comment]
TODO: remove NetworkListener.h before merging!

**Comment by skaupper** [comment]
To review: build the docu using `make doc` and open the `index.html` in docs/ using your browser.

File: src/NetworkListener.h [hunk 1]
```diff
@@ -0,0 +1,90 @@
+/** \file NetworkListener.h
+ *  @brief The interface used by the core logic to get notified of incoming requests.
```
Comment (skaupper) [inline src/NetworkListener.h]: This file does not belong in this PR but it stays for now (as interface reference)

**Comment by wurmmi** [comment]
didn't read doc in detail, but the way the documentation is created is nice :+1: :+1: 

LGTM

**Comment by wurmmi** [comment]
Since #7 was merged, this PR needs to be rebased. Should be easy though..

Please make sure to **SQUASH & MERGE** :slightly_smiling_face:
